spring-data-microservice-transaction-sample/README.md

```diff
@@ -0,0 +1,548 @@
+
+### Schema
+
+[The schema](schema.sql) is as follows:
+
+```sql
+CREATE COORDINATOR TABLES IF NOT EXIST;
+
+CREATE NAMESPACE IF NOT EXISTS customer_service;
+
+CREATE TABLE IF NOT EXISTS customer_service.customers (
+  customer_id INT PRIMARY KEY,
+  name TEXT,
+  credit_limit INT,
+  credit_total INT
+);
+
+CREATE NAMESPACE IF NOT EXISTS order_service;
+
+CREATE TABLE IF NOT EXISTS order_service.orders (
+  customer_id INT,
+  timestamp BIGINT,
+  order_id TEXT,
+  PRIMARY KEY (customer_id, timestamp)
+);
+
+CREATE INDEX IF NOT EXISTS ON order_service.orders (order_id);
+
+CREATE TABLE IF NOT EXISTS order_service.statements (
+  order_id TEXT,
+  item_id INT,
+  count INT,
+  PRIMARY KEY (order_id, item_id)
+);
+
+CREATE TABLE IF NOT EXISTS order_service.items (
+  item_id INT PRIMARY KEY,
+  name TEXT,
+  price INT
+);
+```
+
+All the tables are created in the `customer_service` and `order_service` namespaces.
+
+- `customer_service.customers`: a table that manages customers' information
+  - `credit_limit`: the maximum amount of money a lender will allow each customer to spend when using a credit card
+  - `credit_total`: the amount of money that each customer has already spent by using the credit card
+- `order_service.orders`: a table that manages order information
+- `order_service.statements`: a table that manages order statement information
+- `order_service.items`: a table that manages information of items to be ordered
+
+The Entity Relationship Diagram for the schema is as follows:
+
+![ERD](images/ERD.png)
+
+### Transactions
+
+The following five transactions are implemented in this sample application:
+
+1. Getting customer information. It is a transaction in Customer Service
+2. Placing an order (checks if the cost of the order is below the credit limit, then records order history and updates the `credit_total` if the check passes). It is a transaction that spans Order Service and Customer Service.
+3. Getting order information by order ID. It is a transaction in Order Service
+4. Getting order information by customer ID. It is a transaction in Order Service
+5. Repayment (reduces the amount in the `credit_total`). It is a transaction in Customer Service.
+
+### Service Endpoints
+
+The endpoints defined in the services are as follows:
+
+Customer Service:
+
+- getCustomerInfo
+- payment
+- prepare
+- validate
+- commit
+- rollback
+- repayment
+
+Order Service:
+
+- placeOrder
+- getOrder
+- getOrders
+
+The `getCustomerInfo` endpoint of Customer Service is for transaction #1 (Getting customer information).
+
+And the `placeOrder` endpoint of Order Service and the `payment`, `prepare`, `validate`, `commit`, and `rollback` endpoints of Customer Service are for transaction #2 (Placing an order) that spans Order Service and Customer Service.
+Order Service starts the transaction with the `placeOrder` endpoint, which calls the `payment`, `prepare`, `validate`, `commit`, and `rollback` endpoints of Customer Service.
```

Comment:
```suggestion
The Order Service starts the transaction with the `placeOrder` endpoint, which calls the `payment`, `prepare`, `validate`, `commit`, and `rollback` endpoints of the Customer Service.
```

Reply:
Applying this change in https://github.com/scalar-labs/scalardb-samples/pull/40